alf/utils/normalizers.py

```diff
@@ -107,19 +113,26 @@ def _summary(name, val):
                         alf.summary.scalar(name + ".min." + suffix, val.min())
                         alf.summary.scalar(name + ".max." + suffix, val.max())
 
-            def _summarize_all(t, m, m2, path):
+            def _summarize_all(t, m2, path, m=None):
                 if path:
                     path += "."
                 _summary(path + "tensor.batch_min",
-                         _reduce_along_batch_dims(t, m, torch.min))
+                         _reduce_along_batch_dims(t, m2, torch.min))
                 _summary(path + "tensor.batch_max",
-                         _reduce_along_batch_dims(t, m, torch.max))
-                _summary(path + "mean", m)
-                _summary(path + "var", m2 - math_ops.square(m))
-
-            alf.nest.py_map_structure_with_path(_summarize_all, tensor,
-                                                self._mean_averager.get(),
-                                                self._m2_averager.get())
+                         _reduce_along_batch_dims(t, m2, torch.max))
+                if m is not None:
+                    _summary(path + "mean", m)
+                    _summary(path + "var", m2 - math_ops.square(m))
+                else:
+                    _summary(path + "var", m2)
+
+            if self._mean_averager:
+                alf.nest.py_map_structure_with_path(_summarize_all, tensor,
+                                                    self._mean_averager.get(),
+                                                    self._m2_averager.get())
```

Comment:
mean and m2 order is not correct

Reply:
This is a bit messy, maybe we should have py_map_structure_with_path take a function that has "path" as the first argument:
 _summarize_all(path, t, m2, m=None)

Reply:
good point.